dvc/repo/fetch.py

```diff
@@ -100,29 +102,15 @@ def cb(result):
         results.append(result)
 
     odb = self.odb.local
-    try:
-        with external_repo(
-            repo_url, repo_rev, cache_dir=odb.cache_dir
-        ) as repo:
-            root = PathInfo(repo.root_dir)
-            for path in files:
-                path_info = root / path
-                try:
-                    used_objs, _ = repo.used_cache(
-                        [os.fspath(path_info)],
-                        force=True,
-                        jobs=jobs,
-                        recursive=True,
-                    )
-                    cb(repo.cloud.pull(used_objs, jobs))
-                except (NoOutputOrStageError, NoRemoteError):
-                    pass
-                obj = stage(odb, path_info, repo.repo_fs, "md5", jobs=jobs)
-                save(odb, obj, jobs=jobs, download_callback=cb)
-    except CloneError:
-        failed += 1
-        logger.exception(
-            "failed to fetch data for '{}'".format(", ".join(files))
-        )
+    for obj in external_objs:
+        try:
+            save(odb, obj, jobs=jobs, download_callback=cb)
+        except (ObjectError, OSError):
```

Comment:
`OSError`s during `odb.add` are equivalent to the old fetch/push/pull upload & download errors.